splice_machine/src/main/java/com/splicemachine/derby/stream/function/NLJoinFunction.java

```diff
@@ -129,47 +129,54 @@ public boolean hasNext() {
         try {
             if (rightSideNLJIterator == null)
                 return false;
-            while (nLeftRows > 0 && !rightSideNLJIterator.hasNext()) {
-
-                // We have consumed all rows from right side iterator, reclaim operation context
-                currentOperationContext.getOperation().close();
-                operationContextList.add(currentOperationContext);
-
-                if (leftSideIterator.hasNext()) {
-                    // If we haven't consumed left side iterator, submit a task to scan righ side
-                    ExecRow execRow = leftSideIterator.next();
-                    GetNLJoinIterator getNLJoinIterator = GetNLJoinIterator.makeGetNLJoinIterator(joinType,
-                            operationContextList.remove(0), execRow.getClone());
-                    completionService.submit(getNLJoinIterator);
-                    nLeftRows++;
-                }
+            while (true) {
```

Comment:
Why is this loop needed?

Reply:
When we get a match from the SSQ, I want to skip the remaining matching rows, and let  rightSideNJIterator iterate to the end.

Reply:
I see it now, thanks!